tests/utils/test_async_logging_queue.py

```diff
@@ -0,0 +1,204 @@
+import random
+import threading
+import time
+import uuid
+
+from mlflow.entities.metric import Metric
+from mlflow.entities.param import Param
+from mlflow.entities.run_tag import RunTag
+from mlflow.utils.async_utils.async_logging_queue import AsyncLoggingQueue
+
+METRIC_PER_BATCH = 250
+TAGS_PER_BATCH = 1
+PARAMS_PER_BATCH = 1
+TOTAL_BATCHES = 5
+
+
+class RunData:
+    def __init__(self, throw_exception_on_batch_number=[]) -> None:
+        self.received_run_id = ""
+        self.received_metrics = []
+        self.received_tags = []
+        self.received_params = []
+        self.batch_count = 0
+        self.throw_exception_on_batch_number = (
+            throw_exception_on_batch_number if throw_exception_on_batch_number else []
+        )
+
+    def consume_queue_data(self, run_id, metrics, tags, params):
+        self.batch_count += 1
+        self.received_run_id = run_id
+        self.received_metrics += metrics or []
+        self.received_params += params or []
+        self.received_tags += tags or []
+
+        if self.batch_count in self.throw_exception_on_batch_number:
+            raise Exception("Failed to process batch number: " + str(self.batch_count))
+
+
+def test_single_thread_publish_consume_queue():
```

Comment:
opinion: if you're adding integration tests here anyway (real network calls) maybe add a couple fluent/higher level tests to ensure the full stack doesn't regress with the added benefit of a simple example of usage for customers